src/nacl/bindings/crypto_generichash.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2013 Donald Stufft and individual contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import, division, print_function
+
+from nacl._sodium import ffi, lib
+
+crypto_generichash_BYTES = lib.crypto_generichash_blake2b_bytes()
+crypto_generichash_BYTES_MIN = lib.crypto_generichash_blake2b_bytes_min()
+crypto_generichash_BYTES_MAX = lib.crypto_generichash_blake2b_bytes_max()
+crypto_generichash_KEYBYTES = lib.crypto_generichash_blake2b_keybytes()
+crypto_generichash_KEYBYTES_MIN = lib.crypto_generichash_blake2b_keybytes_min()
+crypto_generichash_KEYBYTES_MAX = lib.crypto_generichash_blake2b_keybytes_max()
+crypto_generichash_SALTBYTES = lib.crypto_generichash_blake2b_saltbytes()
+crypto_generichash_PERSONALBYTES = \
+    lib.crypto_generichash_blake2b_personalbytes()
+
+
+def generichash_blake2b_salt_personal(data,
+                                      digest_size=crypto_generichash_BYTES,
+                                      key=b'', salt=b'', personal=b''):
+    """One time hash interface"""
+    digest = ffi.new("unsigned char[]", digest_size)
+    _salt = ffi.new("char []", crypto_generichash_SALTBYTES)
+    _personal = ffi.new("char []", crypto_generichash_PERSONALBYTES)
+
+    ffi.memmove(_salt, salt, min(len(salt), crypto_generichash_SALTBYTES))
+    ffi.memmove(_personal, personal,
+                min(len(personal), crypto_generichash_PERSONALBYTES)
+                )
+    # _salt and _personal will be truncated or zero-padded
+    # to the correct length
+
```

Comment:
Both parameters are defined as being of a fixed size and zero-filled.
This is the reason why I think the memmove usage to be correct.